cmd/notation/login.go

```diff
@@ -76,55 +78,45 @@ func runLogin(ctx context.Context, opts *loginOpts) error {
 			return err
 		}
 	}
-
 	if opts.Password == "" {
 		opts.Password, err = readPasswordFromPrompt(reader)
 		if err != nil {
 			return err
 		}
 	}
+	cred := opts.Credential()
 
-	if err := validateAuthConfig(ctx, opts, serverAddress); err != nil {
-		return err
+	credsStore, err := auth.NewCredentialsStore()
+	if err != nil {
+		return fmt.Errorf("failed to get credentials store: %v", err)
 	}
-
-	nativeStore, err := auth.GetCredentialsStore(ctx, serverAddress)
+	registry, err := getRegistryClient(ctx, &opts.SecureFlagOpts, serverAddress)
 	if err != nil {
-		return fmt.Errorf("could not get the credentials store: %v", err)
+		return fmt.Errorf("failed to get registry client: %v", err)
 	}
+	if err := credentials.Login(ctx, credsStore, registry, cred); err != nil {
+		registryName := registry.Reference.Registry
+		if !errors.Is(err, credentials.ErrPlaintextPutDisabled) {
+			return fmt.Errorf("failed to log in to %s: %v", registryName, err)
+		}
 
-	// init creds
-	creds := newCredentialFromInput(
-		opts.Username,
-		opts.Password,
-	)
-	if err = nativeStore.Store(serverAddress, creds); err != nil {
-		return fmt.Errorf("failed to store credentials: %v", err)
+		// ErrPlaintextPutDisabled returned by Login() indicates that the credential is validated
+		// but is not saved because there is no native credentials store available
+		if savedCred, err := credsStore.Get(ctx, registryName); err == nil && savedCred == cred {
+			// there is an existing identical credential, ignore saving error
+			fmt.Fprintf(os.Stderr, "Warning: The credentials store is not set up. It is recommended to configure the credentials store to securely store your credentials. See %s.\n", urlDocHowToAuthenticate)
+			fmt.Println()
```

Comment:
The new line seems not necessary?

Reply:
It's to separate the warning and the message

Reply:
But the warning goes to `stderr` and others go to `stdout`. They are two different channels.

Reply:
Yeah but from human's point of view they are displayed together.🤔

Reply:
Removed the new line.